library/std/src/sync/mpmc/list.rs

```diff
@@ -547,6 +547,9 @@ impl<T> Channel<T> {
         }
 
         let mut head = self.head.index.load(Ordering::Acquire);
+        // the channel may be uninitialized, so we have to swap to avoid overwriting any sender's attempts
+        // to initalize the first block before noticing that the receivers disconnected. late allocations
+        // will be deallocated by the sender in Drop
```

Comment:
```suggestion
        // The channel may be uninitialized, so we have to swap to avoid overwriting any sender's attempts
        // to initalize the first block before noticing that the receivers disconnected. Late allocations
        // will be deallocated by the sender in Drop.
```